### ISSUE TITLE
custom [effect]s (GNA #23914)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1511]
**[Original submission](https://gna.org/bugs/index.php?23914)** by anonymous on 2015-09-30

It should be possible to create custom [effect]s (apply_to).

A possible syntax to create new [effect]s could be this:

```

function wesnoth.effects.set_variable(unit, cfg)

 unit.variables[cfg.name] = cfg.value

end

```

which can then be used like this:

```

[effect]

 apply_to = set_variable

 name = varname

 value = 5

[/effect]

```

(Reproduced on )
Release: 1.13
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by wesnoth-bugs** [comment]
**Modified on 2015-09-30**

**anonymous wrote:**

> Also useful, but easier to implement could be a apply_to#lua function<lua_function> [effect]

**Comment by wesnoth-bugs** [comment]
**Modified on 2015-10-02**

**gfgtdf wrote:**

> I tried to implemnt this but it turned out to be quiet hard to do so, the problem is that 'lua proxy units' are expected to be eigher:
> 
> 1)On the map
> 
> 2)On the recall list
> 
> 3)Private lua units
> 
> 
> 
> Also [effect]s are usually applied when the unit is constructed for example via [unit] or [unstore_unit]
> 
> Also units are usually constructed first, and then placed on the map.
> 
> 
> 
> Altogether this means that it is usually not possible to use 'lua proxy units' while [effect]s are applied (when the unit is created). So i think implementing this feature would require rather big changes to how 'lua proxy units' or the unit map in gerneral works.

**Comment by wesnoth-bugs** [comment]
**Modified on 2015-10-02**

**gfgtdf wrote:**

> Well actually not that ai thougth about it again it might be possible to add some 'local c++ variable' variant to lua proxy units, will try.

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-02-07**

**gfgtdf wrote:**

> I implemented this in 1.13.2 It's possible that the game crashes when you remove the unit that currently advanced to the map in  a sustom effect though, it would be nice if we could lock the gamemap (making all gamemap changing lua functions fail) during while the custom effect is applied.

**Comment by CelticMinstrel** [comment]
Since this has been implemented, I'm closing it.
